src/relayclient/RelayClient.ts

```diff
@@ -42,68 +40,72 @@ export interface RelayingResult {
   relayingErrors: Map<string, Error>
 }
 
+export interface TmpDependencyTree {
+  httpClient: HttpClient
+  contractInteractor: ContractInteractor
+  knownRelaysManager: KnownRelaysManager
+  accountManager: AccountManager
+  transactionValidator: RelayedTransactionValidator
+  pingFilter: PingFilter
+  asyncApprove: AsyncApprove
+}
+
 export default class RelayClient {
   private readonly config: RelayClientConfig
-  private readonly web3: Web3
   private readonly httpClient: HttpClient
   private readonly contractInteractor: ContractInteractor
   private readonly knownRelaysManager: KnownRelaysManager
   private readonly accountManager: AccountManager
   private readonly relayHub: Address
   private readonly asyncApprove: AsyncApprove
-  private readonly relayedTransactionValidator: RelayedTransactionValidator
+  private readonly transactionValidator: RelayedTransactionValidator
   private readonly pingFilter: PingFilter
 
   /**
    * create a RelayClient library object, to force contracts to go through a relay.
-   * @param web3  - the web3 instance to use.
-   * @param httpClient
-   * @param contractInteractor
-   * @param knownRelaysManager
-   * @param accountManager
-   * @param relayedTransactionValidator
-   * @param pingFilter
-   * @param config options
-   * @param relayHub
-   * @param asyncApprove
    */
   constructor (
-    web3: Web3,
-    httpClient: HttpClient,
-    contractInteractor: ContractInteractor,
-    knownRelaysManager: KnownRelaysManager,
-    accountManager: AccountManager,
-    relayedTransactionValidator: RelayedTransactionValidator,
+    dependencyTree: TmpDependencyTree,
```

Comment:
bad name. if its the RelayClient, why its args called "Tmp"?

Reply:
I do not like this solution. I want it to have an annoying name :-)